test/api/core/models/repositories/profileRepository.test.js

```diff
@@ -408,6 +408,18 @@ describe('Test: repositories/profileRepository', () => {
         });
     });
 
+    it('should throw a NotFoundError when trying to write unexisting role in profile. ', () => {
+      const invalidProfile = new Profile();
+      invalidProfile._id = 'awesomeProfile';
+      invalidProfile.policies = [{roleId: 'notSoAwesomeRole'}];
+
+      kuzzle.repositories.role.loadRoles = sinon.stub().rejects();
+      return profileRepository.validateAndSaveProfile(invalidProfile)
+        .catch(e => {
+          should(e).be.instanceOf(NotFoundError, { message: 'Unable to hydrate the profile awesomeRole: missing role(s) in the database'});
```

Comment:
Shouldn't this be "unable to hydrate the profile awesomeProfile"?

Reply:
If so, I don't get why this test passes. You probably should add a `.then` handler throwing an exception, just in case the function resolves when it shouldn't

Reply:
Or you can use `rejectedWith`
```
const promise = profileRepository.validateAndSaveProfile(invalidProfile);

return should(promise).be.rejectedWith(NotFoundError, {
  message: 'Unable to hydrate the profile awesomeRole: missing role(s) in the database'
});
```